bubble_sort.rb

```diff
@@ -0,0 +1,37 @@
+def bubble_sort(arr)
+  l = arr.length
+  return arr if l <= 1
+
+  change = true
+
+  while change
+    i = 0
+    while i < l - 1
+      if arr[i] > arr[i + 1]
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      else
+        change = false
+      end
+      i += 1
+    end
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  l = arr.length
+  change = true
+
+  while change
+    i = 0
+    while i < l - 1
+      if (yield arr[i], arr[i + 1]).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      else
+        change = false
+      end
+      i += 1
+    end
+  end
+  arr
+end
```

Comment:
- [ ] Bubble_sort_by should sort an array by accepting a block
For example, if I pass
```
   (bubble_sort_by(unsorted) { |left, right| left.length - right.length }).to eq %w[h hi hey hiya hello]
```
 I should get    
```
       expected: ["h", "hi", "hey", "hiya", "hello"]
```

Reply:
Done!